### PR TITLE
fix: upgrade e2e tests for cdkv2

### PR DESCRIPTION
**Pull request by akshbhu**
#### Description of changes

- Migrate e2e tests package for cdkV2
- fixes geo snapshot tests using prepareApp implementation from CDK

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies
- [ ] [Pull request labels](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#labels) are added

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-category-geo/src/service-utils/prepare-app.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+/**
+ * Prepares the app for synthesis. This function is called by the root `prepare`
+ * (normally this the App, but if a Stack is a root, it is called by the stack),
+ * which means it's the last 'prepare' that executes.
+ *
+ * It takes care of reunifying cross-references between stacks (or nested stacks),
+ * and of creating assets for nested stack templates.
+ */
+export const prepareApp = (root: IConstruct): void => {
```
Comment (sobolk) [inline packages/amplify-category-geo/src/service-utils/prepare-app.ts:17]: I suggest to:
- rename this from `prepareApp` to `cdkPrepareAppShim` or even `cdkv1PrepareAppShim` to discourage usage
- consider marking it deprecated and point people towards cdk.App as alternative
- I'd rename file from `prepare-app.ts` to `cdkv1-compat.ts`

**Review by sobolk — COMMENTED** [review]
LGTM.

I left few comments to make copy of prepareApp less atractive to spread around.

**Review by sobolk — APPROVED** [review]

File: yarn.lock [hunk 2]
```diff
@@ -21403,13 +20995,6 @@ semver@^7.0.0, semver@^7.1.1, semver@^7.3.4, semver@^7.3.5, semver@^7.3.7:
   dependencies:
     lru-cache "^6.0.0"
 
-semver@^7.3.8:
```
Comment (akshbhu) [inline yarn.lock:21406]: checking why this got removed

File: yarn.lock [hunk 3]
```diff
@@ -868,22 +535,6 @@
...
-    jsonschema "^1.4.1"
-    semver "^7.3.7"
-
-"@aws-cdk/cloud-assembly-schema@1.177.0":
-  version "1.177.0"
-  resolved "https://registry.yarnpkg.com/@aws-cdk/cloud-assembly-schema/-/cloud-assembly-schema-1.177.0.tgz#81483d295bd5794c1f2a934bf5b03d7bb4eb21ad"
-  integrity sha512-g1R6lhb5X4xI/rJ6XPJwZF3PwAJVvjkKd+2BYjY6vUD+vmoBIzqIVRVsJX+3MXnqSl0QHYOUstcCqcnqDkNgaA==
-  dependencies:
-    jsonschema "^1.4.1"
-    semver "^7.3.8"
```
Comment (jhockett) [inline yarn.lock:885]: The removed version was depended on here @akshbhu

**Review by jhockett — APPROVED** [review]

**Pull request merged**